tools/codegen/src/main.rs

```diff
@@ -561,6 +636,35 @@ fn download_github(url: &str) -> Result<ureq::Response> {
     Err(last_error.unwrap().into())
 }
 
+#[allow(clippy::missing_panics_doc)]
+pub fn github_head(url: &str) -> Result<()> {
+    eprintln!("fetching head of {url} ..");
```

Comment:
this & the eprintln for failure is a bit noisy, especially as the license algorithm expects this to fail quite often as it tries to detect where to find the license file.